lib/types/config/ServerConfiguration.ts

```diff
@@ -0,0 +1,214 @@
+import { JSONObject } from '../../../index';
+
+
+export type ServerConfiguration = {
+  /**
+   * The maximum size of an incoming request.
+   *
+   * Units can be expressed in bytes ("b" or none), kilobytes ("kb"),
+   * megabytes ("mb"), gigabytes ("gb") or terabytes ("tb").
+   *
+   * @default "1mb"
+   */
+  maxRequestSize: string;
+
+  /**
+  * The listening port for HTTP and WebSocket protocols.
+  *
+  * @default 7512
+  */
+  port: number;
+
+  /**
+  * Configuration section for Kuzzle access logs.
+  */
+  logs: {
+    /**
+    * An array of Winston transports configurations to output access
+    * logs.
+    *
+    * Possible transport types are: console, file, elasticsearch and syslog.
+    *
+    * Please refer to https://github.com/winstonjs/winston/blob/master/docs/transports.md
+    * for more information on transports configuration.
+    *
+    * @default
+    *
+    * [
+        {
+          transport: 'console',
+          level: 'info',
+          stderrLevels: [],
+          silent: true
+        }
+      ]
+    *
+    */
+    transports: JSONObject[];
+
+    /**
+    * Access log format.
+    *
+    * Currently supported are "combined" (=Apache combined logs format)
+    * and "logstash".
+    *
+    * "logstash" will output the whole request input to JSON, ready to
+    * be consumed by logstash agent.
+    *
+    * @default "combined"
+    */
+    accessLogFormat: string;
+
+    /**
+    * The offset to use as the client ip, from the FORWARDED-FOR chain,
+    * beginning from the right (0 = the ip address of the last
+    * client|proxy which connected to Kuzzle.
+    *
+    * @default 0
+    */
+    accessLogIpOffset: number
+  },
+
+  /**
+  * protocols accepted by Kuzzle. 
+  * protocols can be extended and configured in this section.
+  */
+  protocols: {
+    http: {
+      /**
+      * Enable support for compressed requests, using the Content-Encoding header 
+      * Currently supported compression algorithms:  gzip, deflate, identity 
+      * Note: "identity" is always an accepted value, even if compression support is disabled
+      * 
+      * @default true
+      */
+      allowCompression: boolean
+
+      /**
+       * Set to "false" to disable HTTP support
+       * 
+       * @default true
+       */
+        enabled: true
```

Comment:
Is this always `true` or `boolean` ? Also you have an indentation problem starting here